### PR TITLE
add typings for js-sha1 0.6

### PR DESCRIPTION
**Pull request by yhaskell**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by ehirina — APPROVED** [review]

File: types/js-sha1/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    update(message: string): string;
+    hex(message: string): string;
+    array(message: string): number[];
+    digest(message: string): number[];
+    arrayBuffer(message: string): ArrayBuffer;
+}
+
+declare var sha1: JsSha1;
+
+export = sha1;
```
Comment (peterblazejewicz) [inline types/js-sha1/index.d.ts:17]: this one looks like old-school UMD module, not CommonJS module:
https://github.com/emn178/js-sha1/blob/master/src/sha1.js
so: https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#identifying-a-umd-library
Please review the module type

**Review by gabritto — CHANGES_REQUESTED** [review]

File: types/js-sha1/index.d.ts (hunk 1 above)
Comment (yhaskell) [reply types/js-sha1/index.d.ts:17]: the module works as CommonJS and UMD module (see https://github.com/emn178/js-sha1/blob/65b3898c4eecc919477dce9c6e41bb0be1b65219/src/sha1.js#L361). 
I've added an `export as namespace` following @gabritto's advice, so the `export =` also stays, right?

**Review by ehirina — APPROVED** [review]

**Review by gabritto — APPROVED** [review]

File: types/js-sha1/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/js-sha1/index.d.ts:17]: yes, thx!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@yhaskell thanks!

**Comment by yhaskell** [comment]
Ready to merge

**Pull request merged**